src/deploy/osp_deployer/settings_sanity.py

```diff
@@ -489,3 +489,54 @@ def verify_iha_dependency_on_fencing(self):
     def validate_profile(self):
         self.profile = Profile()
         self.profile.validate_configuration()
+
+    def validate_nic_configs(self):
+
+        # Get the user supplied NIC settings from the .ini
+        ini_nics_settings = self.settings.get_nics_settings()
+
+        missing_settings = []
+        nic_env_file_missing = False
+        if 'nic_env_file' not in ini_nics_settings:
+            nic_env_file_missing = True
+            missing_settings.append('nic_env_file')
+
+        if 'sah_bond_opts' not in ini_nics_settings:
+            missing_settings.append('sah_bond_opts')
+
+        extra_settings = set()
+        missing_variable_settings = set()
+        if not nic_env_file_missing:
```

Comment:
if nic_env_file is missing from the ini file - it will return an error with only that setting missing logged (even tho you might have additional ones missing)
Would be best to also check the rest of the settings so all are reported in one shot.

Reply:
It will report at least on the status of both nic_env_file and sah_bond_opts if one or the other or both are missing.
Because the remaining NIC configuration parameters will be different depending on the NIC configuration files, we have to validate the remaining NIC configuration settings in the user's .ini file with the NIC configuration settings in the nic_env_file.  As a result, if nic_env_file is missing then we can't validate the remaining NIC configuration settings.